pandas/tests/indexes/test_multi.py

```diff
@@ -1601,6 +1601,19 @@ def test_from_tuples(self):
         idx = MultiIndex.from_tuples(((1, 2), (3, 4)), names=['a', 'b'])
         self.assertEqual(len(idx), 2)
 
+    def test_from_tuples_variable_length(self):
+        # check that len(MultiIndex) == max(len(iterables))
+        T = ((1,), (2, 3), (4, 5, 6))
```

Comment:
I know from_arrays required balanced inputs (there's a check for it).  It seems that from_tuples attempts to balance the input before passing it to from_arrays.